nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/main/java/org/apache/nifi/web/security/spring/LoginIdentityProviderFactoryBean.java

```diff
@@ -219,16 +217,14 @@ private LoginIdentityProviderConfigurationContext loadLoginIdentityProviderConfi
     }
 
     private String decryptValue(String cipherText, String encryptionScheme) throws SensitivePropertyProtectionException {
-            initializeSensitivePropertyProvider(encryptionScheme);
+        initializeSensitivePropertyProvider(encryptionScheme);
         return SENSITIVE_PROPERTY_PROVIDER.unprotect(cipherText);
     }
 
     private static void initializeSensitivePropertyProvider(String encryptionScheme) throws SensitivePropertyProtectionException {
-        if (SENSITIVE_PROPERTY_PROVIDER == null || !SENSITIVE_PROPERTY_PROVIDER.getIdentifierKey().equalsIgnoreCase(encryptionScheme)) {
+        if (SENSITIVE_PROPERTY_PROVIDER == null) {
             try {
-                String keyHex = getMasterKey();
-                SENSITIVE_PROPERTY_PROVIDER_FACTORY = new AESSensitivePropertyProviderFactory(keyHex);
-                SENSITIVE_PROPERTY_PROVIDER = SENSITIVE_PROPERTY_PROVIDER_FACTORY.getProvider();
+                SENSITIVE_PROPERTY_PROVIDER = StandardSensitivePropertyProvider.fromKey(getMasterKey(), encryptionScheme);
```

Comment:
May need to write test cases around a `login-identity-providers.xml` file with the AWS KMS key ID embedded in the `encryption="..."` attribute of the individual encrypted elements.

Reply:
I think we should add this type of test, but I'm not sure how to expose the AWS test classes across packages.  I'll open the a ticket to track this for now.

Reply:
Added note about these tests to NIFI-6402.